map/local_ads_helpers.cpp

```diff
@@ -0,0 +1,92 @@
+#include "map/local_ads_helpers.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+#include "base/bits.hpp"
+
+namespace
+{
+template<typename TPrimitive>
+TPrimitive ReadPrimitive(FileReader const & reader, uint64_t & offset)
+{
+  if (offset + sizeof(TPrimitive) > reader.Size())
+    MYTHROW(Reader::SizeException, ("File is corrupted", offset, sizeof(TPrimitive), reader.Size()));
+
+  std::vector<char> bytes(sizeof(TPrimitive));
```

Comment:
I suggest you to use ReaderSource + ReadPrimitiveFromSource, instead of this manipulations with byte buffers.